server/cluster/cluster.go

```diff
@@ -1277,23 +1278,26 @@ func (c *RaftCluster) UpStore(storeID uint64) error {
 		return nil
 	}
 
-	limiter := c.prevStoreLimit[storeID]
 	options := []core.StoreCreateOption{core.UpStore()}
-	// To prevent panic we check if the store limit is nil. There are two cases may cause this problem:
-	// 1. when we start a new cluster, the store limit is only initialized for the first time scheduling
-	// 2. once the PD leader transfer, the previous store limit may be lost, we suggest changing it manually
-	if limiter[storelimit.AddPeer] != nil {
-		options = append(options, core.ResetStoreLimit(storelimit.AddPeer, limiter[storelimit.AddPeer].Rate()))
-	}
-	if limiter[storelimit.RemovePeer] != nil {
-		options = append(options, core.ResetStoreLimit(storelimit.RemovePeer, limiter[storelimit.RemovePeer].Rate()))
+	// get the previous store limit recorded in memory
+	limiter, exist := c.prevStoreLimit[storeID]
+	if exist {
+		options = append(options,
+			core.ResetStoreLimit(storelimit.AddPeer, limiter[storelimit.AddPeer]),
```

Comment:
Don't we need to check whether the `limiter[storelimit.AddPeer]` is nil anymore?

Reply:
If the store limit has existed, add peer won't be nil.